cli/cli.ts

```diff
@@ -4084,212 +4115,263 @@ function testSnippetsAsync(snippets: CodeSnippet[], re?: string, pyStrictSyntaxC
-                        return addSuccess(fn)
-                    }
+            // decompile to python
+            const decompiled = pxtc.service.performOperation("pydecompile", {
+                options: opts,
+                fileName: pxt.MAIN_TS,
+            }) as pxtc.transpile.TranspileResult;
+
+            const py = decompiled.outfiles[pxt.MAIN_PY];
+            const pySuccess = !!py && decompiled.success;
+            if (!pySuccess) {
+                console.log("ts2py error");
+                return addFailure(snippet.src, decompiled.diagnostics);
+            }
+            opts.fileSystem[pxt.MAIN_PY] = py;
+
+            // py to ts
+            const ts1 = opts.fileSystem["main.ts"];
+            opts.target.preferredEditor = pxt.PYTHON_PROJECT_NAME;
+            let ts2Res = pxtc.service.performOperation("py2ts", {
+                options: opts,
+            }) as pxtc.transpile.TranspileResult;
+
+            let ts2 = ts2Res.outfiles["main.ts"];
+
+            if (!ts2) {
+                console.log("py2ts error!");
+                console.dir(ts2Res);
+                let errs = ts2Res.diagnostics.map(pxtc.getDiagnosticString).join();
+                if (errs)
+                    console.log(errs);
+                return addFailure(snippet.src, ts2Res.diagnostics);
+            }
+
+            if (pyStrictSyntaxCheck) {
+                let cmp1 = pyComparisonString(ts1);
+                let cmp2 = pyComparisonString(ts2);
+                if (cmp1 != cmp2) {
+                    console.log(`Mismatch. Original:`);
+                    console.log(cmp1);
+                    console.log("decompiled->compiled:");
+                    console.log(cmp2);
+                    console.log("TS mismatch :/");
+                    // TODO: generate more helpful diags
+                    return addFailure(snippet.src, []);
+                } else {
+                    console.log("TS same :)");
                 }
-                else {
-                    return addFailure(name, resp.diagnostics)
+            }
+
+            // NOTE: neither of these decompile steps checks that the resulting code is correct or that
+            // when the code is compiled back to ts it'll behave the same. This could be validated in
+            // the future.
+
+            return addSuccess(name)
+        } catch (e) {
+            console.dir(e);
+            addFailure(
+                snippet.src,
+                [createKsDiagnostic(e.message, fn)]
+            );
+        }
+    });
+
+    pxt.log(`${successes.length}/${successes.length + failures.length} snippets compiled to blocks and python (and back), ${failures.length} failed`);
+    if (ignoreCount > 0) {
+        pxt.log(`Skipped ${ignoreCount} snippets`);
+    }
+    if (failures.length > 0) {
+        const msg = `${failures.length} snippets not compiling in the docs`
+        pxt.log(`${msg}:`);
+
+        failures.forEach(failure => {
```

Comment:
I found it annoying looking inline for the errors / having them trimmed / not listing the origin file, so I made it read them out in detail at the end